api/src/main/java/org/openmrs/module/fhir/api/impl/ConditionServiceImpl.java

```diff
@@ -49,27 +48,17 @@ public void setDao(FHIRDAO dao) {
 
 	@Override
 	public Condition getCondition(String id) {
-		org.openmrs.Condition condition = Context.getService(org.openmrs.api.ConditionService.class).getConditionByUuid(id);
-		if (condition == null || condition.isVoided()) {
-			return null;
-		}
-		return FHIRConditionUtil.generateFHIRCondition(condition);
+		return ContextUtil.getConditionHelper().getCondition(id);
 	}
 
 	@Override
 	public Condition getConditionByObsId(String id) {
-		return ConditionStrategyUtil.getConditionStrategy().getConditionById(id);
+		return ConditionStrategyUtil.getConditionStrategy().getConditionByObsId(id);
 	}
 
 	@Override
 	public List<Condition> searchConditionById(String id) {
-		org.openmrs.Condition condition = Context.getService(org.openmrs.api.ConditionService.class)
-				.getConditionByUuid(id);
```

Comment:
Why did you replace this with null?

Reply:
It was the previous implementation, not working as expected.
Will redo in https://issues.openmrs.org/browse/FM-65

Reply:
Shouldn't we throw NotImplementedException instead of returning null?

Reply:
Did you see the above?